src/blueprint/blueprint.rs

```diff
@@ -219,4 +221,53 @@ impl Blueprint {
     // generation of schema cannot fail.
     schema.finish().unwrap()
   }
+
+  /// Generate a dumb schema useful for validation.
+  pub fn to_validation_schema(&self) -> Schema {
+    create(self, Some(SchemaModifiers { no_resolver: true }))
+      .validation_mode(ValidationMode::Strict)
+      .disable_introspection()
+      .finish()
+      .unwrap()
+  }
+
+  pub async fn validate_operation(schema: &Schema, query: &str) -> Vec<Cause<String>> {
+    schema
+      .execute(query)
+      .await
+      .errors
+      .iter()
+      .map(|err| {
+        Cause::new(err.message.clone()).trace(VecDeque::from_iter(
+          err
+            .locations
+            .iter()
+            .map(|loc| format!("line {} column {}", loc.line, loc.column)),
+        ))
+      })
+      .collect()
+  }
+
+  pub async fn validate_operations(&self, operations: Vec<String>) -> Valid<(), String> {
+    let schema = self.to_validation_schema();
+    let mut execution = vec![];
+
+    for op in operations.iter() {
+      match tokio::fs::read_to_string(op).await {
+        Ok(operation) => {
+          let causes = Blueprint::validate_operation(&schema, &operation).await;
+          execution.push((op.to_string(), causes));
+        }
+        Err(_) => execution.push((
+          op.to_string(),
+          vec![Cause::new(format!("Cannot read file operation file {}", op))],
+        )),
+      }
+    }
+
+    Valid::from_iter(execution.iter(), |(op, causes)| {
+      Valid::<(), String>::from_vec_cause(causes.to_vec()).trace(op)
+    })
+    .map(|_: Vec<_>| ())
```

Comment:
why do we need this?

Reply:
It converts the errors from all operations into one, and discard the expected `Vec<()>` for `()`.

Reply:
That allows to display a structured output with each failure linked to the appropriate file.